### PR TITLE
GLTFExporter: Add forcePowerOfTwoTexture option

### PR DESCRIPTION
**Pull request by takahirox**
https://github.com/mrdoob/three.js/issues/13397#issuecomment-368256557

This PR adds `forcePowerOfTwoTexture` option to `GLTFExporter`.

### CODE REVIEW

File: examples/js/exporters/GLTFExporter.js [hunk 1]
```diff
@@ -459,21 +473,36 @@ THREE.GLTFExporter.prototype = {
...
 
 				var canvas = cachedCanvas = cachedCanvas || document.createElement( 'canvas' );
+
 				canvas.width = map.image.width;
 				canvas.height = map.image.height;
+
+				if ( options.forcePowerOfTwoTexture ) {
+
+					if ( ! options.embedImages ) {
+
```
Comment (mrdoob) [inline examples/js/exporters/GLTFExporter.js]: I don't understand this check. We're showing the warning only if `embedImages` is `false`?

File: examples/js/exporters/GLTFExporter.js [hunk 2]
```diff
@@ -459,21 +473,36 @@ THREE.GLTFExporter.prototype = {
...
 
-			if ( options.embedImages ) {
+			if ( options.embedImages || ( options.forcePowerOfTwoTexture && ! isPowerOfTwo( map.image ) ) ) {
 
 				var canvas = cachedCanvas = cachedCanvas || document.createElement( 'canvas' );
+
 				canvas.width = map.image.width;
 				canvas.height = map.image.height;
+
+				if ( options.forcePowerOfTwoTexture ) {
```
Comment (mrdoob) [inline examples/js/exporters/GLTFExporter.js]: Wouldn't it make more sense to move the previous check here?

```js
if ( options.forcePowerOfTwoTexture && ! isPowerOfTwo( map.image ) ) {
```

So we only do the resize when `embedImages` is `true` , and if we do the resize we show the warning...

File: examples/js/exporters/GLTFExporter.js (hunk 1 above)
Comment (takahirox) [reply examples/js/exporters/GLTFExporter.js]: In that case, user doesn't set `embedImages` true so expects non-embedded image., but image will be embedded so I wanna notify it to users.

**Comment by mrdoob** [comment]
image

Uh? I didn't. I deleted my own comments...

File: examples/js/exporters/GLTFExporter.js [hunk 3]
```diff
@@ -459,21 +473,36 @@ THREE.GLTFExporter.prototype = {
 			var mimeType = map.format === THREE.RGBAFormat ? 'image/png' : 'image/jpeg';
 			var gltfImage = { mimeType: mimeType };
 
-			if ( options.embedImages ) {
+			if ( options.embedImages || ( options.forcePowerOfTwoTexture && ! isPowerOfTwo( map.image ) ) ) {
```
[the remark this answers is not in the document]
Comment (takahirox) [reply examples/js/exporters/GLTFExporter.js]: So, you meant you wanna POT conversion only if `options.embedImages` is true?

**Comment by takahirox** [comment]
> Uh? I didn't. I deleted my own comments...

Haha, ok I believe in you.

File: examples/js/exporters/GLTFExporter.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrdoob) [reply examples/js/exporters/GLTFExporter.js]: Yeah... At least for now.

[the remark this answers is not in the document]
Comment (takahirox) [reply examples/js/exporters/GLTFExporter.js]: OK, I've updated. We might need to make clear that `.forcePowerOfTwoTexture` option works only if `.embedImages` is true by renaming or adding note to doc. Let's do that in another PR if necessary (sorry time to sleep here now!).

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**